### ISSUE TITLE
Standardize logging

### ISSUE THREAD

**Issue by jameshiebert** [pacificclimate/crmprtd#20]
Now that we're actually [doing something with our logs](https://www.elastic.co/elk-stack), it will become important for them to have a bit more structure than the free flowing, laissez faire approach that we have taken to date.

1. Logs should be *parseable*: we need to use a log formatter that can be easily consumed by logstash and can handle multiline messages and stacktraces. Most advice says to use a JSON log formatter.
1. Logs should provide *metrics*: data points that get inserted vs. skipped (due to being duplicates) vs. errors are all extremely important metrics for this package. Provide these values as specifically attributed values in the log entries at the conclusion of each run. Insertions / second gives us good information on database code performance. We can get creative on measuring other things, but these are the most important.
1. Logs (and information levels) should be *consistent*. Right now are logs are a little scattered about what type of information belongs at which level of importance (DEBUG, INFO, WARNING, etc.). We should define the levels at which certain information should be and make that consistent across all of our modules.

We're already using [Python's logging module](https://docs.python.org/3/library/logging.html) which will make this relatively easy. We just need to define an application-wide logging config and use it consistently across modules. There are only 99 logging statements in the code, so this shouldn't be too bad.

~~~bash
james@basalt:~/code/git/crmprtd/crmprtd$ grep "log\." *.py | wc -l
99
~~~

Development wise, we should work on this as a hotfix branch off of master.

**Comment by nikola-rados** [comment]
Sounds like a good idea. I'll get on it!

**Comment by jameshiebert** [comment]
Further to this issue, we need to ensure that all of the scripts use the same approach (and code) to logging configuration. They should each optionally accept (and *use*) a YAML config file that can override sane logging defaults.

**Comment by nikola-rados** [comment]
@jameshiebert while merging the logging changes into ```dev``` I discovered an error I had made.  The ```log_level``` variable isn't being set properly.  I had incorrectly assumed that the level in the **'handlers'** (from ```/data/logging.yaml```) needed to be changed to set the level.  However, I found [this](https://docs.python.org/3/library/logging.html#logging.Logger.setLevel) which says the new logger object will get its level from the parent if it is not set using ```setLevel()```.  I thought the parent would be the **'handler'** but after investigation it turns out it is **'root'**.  This means instead of setting the handler levels I should be changing the root level.  I have already changed this in ```dev``` after the merge, should I merge the hotfix branch into ```master```?
